### DISCUSSION TITLE
Create reloadable database from shell.duckdb.org and connect to DBeaver

### DISCUSSION THREAD

**Discussion by Starlord22** [duckdb/duckdb#7021] [General]
Hello, I'm trying to use duckdb thru shell.duckdb.org, but each time I connect to the database I need to create table and import csv file.

I read here about creating reloadable database in python. https://github.com/duckdb/duckdb/issues/248

1. Any guidance on how to create, persist and commit the tables using shell.duckdb.org
2. Once I am able to persist the db, how can I connect the db instance to DBeaver IDE? steps on how to accomplish this are very much appreciated.

**Comment by Alex-Monahan** [comment]
Howdy! I don't think we have a way to export the entire database today. Persistence is a part of the roadmap though, now that browsers have made that possible.

**Reply by carlopi** [reply]
There is the possibility to download files, even the entire database, so if you control the embedding you can download the relevant state and persist it between sessions, but indeed this is not provided by shell.duckdb.org out of the box. There are plans to enhance persistency browser capabilities, but persistency, expecially being robust to browser cache eviction, will likely not be of the shell deployment for a while.
